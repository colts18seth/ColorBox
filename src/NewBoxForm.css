label {
    display: block;
    margin: .5rem;
}

input {
    margin-left: 1rem;
}