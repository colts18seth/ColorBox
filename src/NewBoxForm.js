import React, { useState } from 'react';
import './NewBoxForm.css'

function NewBoxForm({ addBox }) {
    const INITIAL_STATE = {
        width: "",
        height: "",
        color: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };

    const handleSubmit = e => {
        e.preventDefault();
        addBox(formData.width, formData.height, formData.color);
        setFormData(INITIAL_STATE);
    }

    return (
        <form className="NewBoxForm" onSubmit={handleSubmit}>
            <label htmlFor="width">Width:
            <input
                    id="width"
                    name="width"
                    type="text"
                    value={formData.width}
                    onChange={handleChange}
                />
            </label>
            <label htmlFor="height">Height:
            <input
                    id="height"
                    name="height"
                    type="text"
                    value={formData.height}
                    onChange={handleChange}
                />
            </label>
            <label htmlFor="color">Background Color:
            <input
                    id="color"
                    name="color"
                    type="text"
                    value={formData.color}
                    onChange={handleChange}
                />
            </label>
            <button>Add Box</button>
        </form>
    );
}

export default NewBoxForm;