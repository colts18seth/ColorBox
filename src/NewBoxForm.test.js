import React from 'react';
import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

it('renders without crashing', () => {
    render(<NewBoxForm />);
});

it('matches snapshot', () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

it('renders form fields', () => {
    const { getByText } = render(<NewBoxForm />);
    const addBoxButton = getByText('Add Box');
    const width = getByText('Width:');
    const height = getByText('Height:');
    const backgroundColor = getByText('Background Color:');

    expect(addBoxButton).toBeInTheDocument();
    expect(width).toBeInTheDocument();
    expect(height).toBeInTheDocument();
    expect(backgroundColor).toBeInTheDocument();
})