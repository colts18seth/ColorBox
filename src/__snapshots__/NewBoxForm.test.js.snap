// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="width"
    >
      Width:
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
    </label>
    <label
      for="height"
    >
      Height:
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
    </label>
    <label
      for="color"
    >
      Background Color:
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
    </label>
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
